code-scanning/pysa.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow integrates Python Static Analyzer (Pysa) with
+# GitHub's Code Scanning feature.
+#
+# Python Static Analyzer (Pysa) is a security-focused static
+# analysis tool that tracks flows of data from where they
+# originate to where they terminate in a dangerous location.
+#
+# See https://pyre-check.org/docs/pysa-basics/
+
+name: Pysa
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron: $cron-weekly
+
+jobs:
+  pysa:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Run Pysa
+        uses: facebook/pysa-action
```

Comment:
Typically code scanning template upload sarif file. Is it being uploaded in your action? If not then do you need to add upload step?

Reply:
Pysa Action does upload its results in SARIF:

https://github.com/facebook/pysa-action/blob/f46a63777e59268613bd6e2ff4e29f144ca9e88b/action.yml#L147-L159

The upload is handled by SAPP Action here:

https://github.com/facebook/sapp-action/blob/fbb43085f87b1d1b76d88b8195a76382155ae370/action.yml#L65-L75

Reply:
Thanks for sharing this